### PR TITLE
Add symbolic link resolution to getFileContent.

### LINKED ISSUE #9: Add support for symbolic links
_From @julianharty on May 8, 2017 20:25_

Here's what I know, the bug may be more general or very specific to my case. More testing could help identify the extent, for instance to see if the same behavior happens when using soft-links for images, html, etc.

# Ingredients:

* zimwriterfs (I used the current Docker Image that contains it). Note I added a shared folder so I could easily share data between the running Docker image and my macbook. This had to be configured using the command line. See https://github.com/rocker-org/rocker/wiki/Sharing-files-with-host-machine for tips on how to do this.
* some basic content. I generated it from the https://github.com/julianharty/testing-heuristics project (follow the instructions in the README)
* create a folder called `./A` in the `dist` folder
* create a soft-link using `ln -s` for style.css: `ln -s style.css A/style.css`
* run zimwriterfs with the following command (in the extract below) in the running Docker container.

Using a soft-link (`ln -s source-file destination-file`) caused `zimwriterfs` to report and error and fail as follows:
```
root@3090e980e81e:/# zimwriterfs  --uniqueNamespace  -w index.html -f icon.png -l EN -t "Testing Heuristics" -d "Early cut of the testing heuristics material." -c "Julian Harty" -p "Commercetest Ltd." /Volumes/shared/dist/ /Volumes/shared/th-ln.zim

create directory entries
collect articles
zimwriterfs: unable to open file at path: /Volumes/shared/dist/A/style.css
terminate called after throwing an instance of 'int'
Aborted
```
Copying the file and then re-running the identical `zimwriterfs` command worked (I had to delete the soft-link first).
```
rm A/style.css 
cp style.css A/
```

_Copied from original issue: openzim/libzim#16_

### PR DESCRIPTION
**Pull request by akuendig**
Fixes #9.

### CODE REVIEW

**Comment by kelson42** [comment]
@akuendig Thank you very much for your patch. Be patient, it might take a few days, up to a few weeks to be reviewed and merged.

**Comment by kelson42** [comment]
@mgautierfr Could you please have a look to this PR?

File: src/tools.cpp [hunk 1]
```diff
@@ -155,30 +156,83 @@ inline bool seemsToBeHtml(const std::string& path)
...
+      std::cerr << "zimwriterfs: unable to resolve symlink: " << path
+                << std::endl;
+      throw(errno);
+    }
+    if (link_len > path_stat.st_size) {
+      std::cerr << "zimwriterfs: symbolic link target path increased in size "
+                << "between lstat and readlink: " << path << std::endl;
+      continue;
+    }
+    resolved_path.resize(path_stat.st_size);
```
Comment (mgautierfr) [inline src/tools.cpp]: We should resize to `link_len`. If the target path has changed between `lstat` and `readlink`, the correct value is `link_len`, not `path_stat.st_size`.

**Comment by mgautierfr** [comment]
It appears that this PR is unnecessary, see my comment is #9 .

**Pull request closed without merging**